c/collections.c

```diff
@@ -1417,6 +1417,84 @@ void *qRemove(Queue *q){
 #endif /* END OF OS-VARIANT Queue stuff */
 
 
+/* The Array List (flexible Array thing that smells like Java and javascript) */
+
+static void *arrayListAlloc(ArrayList *list, uint32_t size, char *location){
+  if (list->slh){
+    return (void*)SLHAlloc(list->slh,size);
+  } else{
+    return safeMalloc(size,location);
+  }
+}
+
+ArrayList *makeArrayList(){
+  ArrayList *list = (ArrayList*)safeMalloc(sizeof(ArrayList),"ArrayList");
```

Comment:
Do we want to have NULL checks in library code, so that the caller can decide what to do with OOM?

Reply:
We are probably not consistent about this.   I would prefer safeMalloc and friends plus SLHAlloc should probably log reliably to stderr or WTO

Reply:
Ok, we may not be able to use this in xmem because we try not to ABEND there, and logging may also be an issue. Whereas a bad RC would let the caller know about OOM.

Reply:
General hardening of the common data structures will be done separately.